rust/rope/src/spans.rs

```diff
@@ -331,9 +349,10 @@ impl<'a, T: Clone> Iterator for SpanIter<'a, T> {
     type Item = (Interval, &'a T);
 
     fn next(&mut self) -> Option<(Interval, &'a T)> {
-        if let Some((leaf, start_pos)) = self.cursor.get_leaf() {
+        while let Some((leaf, start_pos)) = self.cursor.get_leaf() {
```

Comment:
The behavior was to bail on the first empty `leaf.spans`. Since we allow removing spans, it is possible for a leaf to have no spans but for spans to follow.

Reply:
`TreeBuilder::push_leaf` has no visibility into leaves other than what their length is. Since the length is the same as before, I believe it will still add leaves with no spans. We have to leave it this way unless we want to add the length to the previous leaf & omit it.